### PR TITLE
Move tests inside the package and ignore them when packaging

### PR DESCRIPTION
**Pull request by jeremycline**
I had a lot of time on a plane recently so I did some reorganizing and tidying.

This places the tests inside the ``anitya`` package and ignores them
when packaging. This allows us to avoid fiddling with the system path in
the tests.

All VCR cassettes have been renamed to match the new test paths.

Signed-off-by: Jeremy Cline <jeremy@jcline.org>

### CODE REVIEW

File: anitya/tests/base.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # -*- coding: utf-8 -*-
 #
-# Copyright © 2014  Red Hat, Inc.
+# Copyright © 2017  Red Hat, Inc.
```
Comment (pypingou) [inline anitya/tests/base.py]: Shouldn't it be 2014-2017?

File: anitya/tests/lib/backends/test_gnome.py [hunk 2]
```diff
@@ -27,13 +27,10 @@
 import sys
 import os
 
-sys.path.insert(0, os.path.join(os.path.dirname(
-    os.path.abspath(__file__)), '..'))
-
```
Comment (pypingou) [inline anitya/tests/lib/backends/test_gnome.py]: sys and os are still imported here while you dropped them from other tests, intended?

File: anitya/tests/lib/backends/test_gnu.py [hunk 3]
```diff
@@ -31,13 +31,10 @@
 import sys
 import os
```
Comment (pypingou) [inline anitya/tests/lib/backends/test_gnu.py]: Same as above :)

File: anitya/tests/lib/backends/test_launchpad.py [hunk 4]
```diff
@@ -31,13 +31,10 @@
 import sys
 import os
```
Comment (pypingou) [inline anitya/tests/lib/backends/test_launchpad.py]: Here as well :)

**Comment by pypingou** [comment]
Couple of questions regarding some import left, otherwise looks fine

File: anitya/tests/base.py (hunk 1 above)
Comment (jeremycline) [inline anitya/tests/base.py]: I think this is the result of an odd diff where I made the new base.py file, then started moving stuff out of __init__.py. In the end I moved pretty much everything so git thinks (fairly, I suppose) I renamed the file and made a new __init__.py. I will adjust it.

**Comment by jeremycline** [comment]
@pypingou I removed all the imports I missed.

@ncoghlan this will affect #401 (although it should be a simply git mv of those two new test modules).

**Comment by pypingou** [comment]
Looks fine to me :)

**Pull request merged**